content/client-lib-development-guide/features.textile

```diff
@@ -107,6 +107,7 @@ h3(#rest-auth). Auth
 *** @(RSA4c1)@An @ErrorInfo@ with @code@ @80019@ and description of the underlying failure should be emitted with the state change, in the @errorReason@ and/or in the callback as appropriate
 *** @(RSA4c2)@If the connection is @CONNECTING@, then the connection attempt should be treated as unsuccessful, and as such the connection should transition to the @DISCONNECTED@ or @SUSPENDED@ state as defined in "RTN14":#RTN14 and "RTN15":#RTN15
 *** @(RSA4c3)@If the connection is @CONNECTED@, then the connection should remain @CONNECTED@
+** @(RSA4d)@ If an attempt to authenticate is made using the @authUrl@ and the content type is neither text nor JSON, or if using @authCallback@ and the object passed in is neither a @String@, @JsonObject@, @TokenRequest@ object nor a @TokenDetails@ object, the auth attempt should fail with error code @40143@ and a suitable error message.  Additionally, if the text token string is greater than 384 bytes, or the JSON stringified @JsonObject@, @TokenRequest@ or @TokenDetails@ is greater than 128kb, then the auth attempt should fail with error code @40143@ and a suitable error message.
```

Comment:
> neither text nor JSON

this should probably read "neither `text/plain` or `application/json`" -- technically `text/html` is a kind of text

> or the JSON stringified `JsonObject`, `TokenRequest` or `TokenDetails` is greater than 128kb

what's this protecting against? tokenrequests and tokendetails are easy to detect heruistically, and if it's a valid one but with an 129k capabilities string, that's admittedly a bit ridiculous, but at that point the client has it already, so why reject it? (An object that isn't a valid tokenDetails or tokenRequest should be rejected already)

> 40143

this is inconsistent with the error codes for other reasons for authCallback and authUrl failing (including the token being in an invalid format), which are specified in RSA4c. (40170 for standalone, 80019 on the realtime connection). 40143 is used by realtime.

How about just bundling this in with RSA4c, which already specifies what to do when " the token is in an invalid format", which this is just clarifying:

`(RSA4c4) A returned token shall be taken to be in an invalid format if: the content-type of the response from @authUrl@ is neither @text/plain@ or @application/json@; the object returned by @authCallback@ is neither a @String@, @JsonObject@, @TokenRequest@ object nor a @TokenDetails@ object; or if the text token string is greater than 384 bytes.`